File: solutions/scala/darts/1/src/main/scala/Darts.scala
import scala.math.sqrt

object Darts {
  def score(x: Double, y: Double): Int = {
    val dist = sqrt(x * x + y * y)
    
    if(dist > 10) 0
    else if(dist > 5) 1
    else if(dist > 1) 5
    else 10
  }
}

File: solutions/clojure/darts/1/src/darts.clj
(ns darts)

(defn score
  "Calculates the score of a dart throw."
  [x y]
  (let [dist (Math/sqrt (+ (* x x) (* y y)))]
    (cond
      (> dist 10) 0
      (> dist 5) 1
      (> dist 1) 5
      :else 10)))
